### PR TITLE
Enhance InsertData and FieldData

### PR DESCRIPTION
**Pull request by XuanYang-cn**
1. Add NewInsertData
2. Add GetRowNum(), GetMemorySize(), and, Append() for InsertData
3. Add AppendRow() for FieldData for compaction

/kind improvement

### CODE REVIEW

File: internal/storage/insert_data.go [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+
+func (data *FloatVectorFieldData) GetRow(i int) interface{} {
+	return data.Data[i*data.Dim : (i+1)*data.Dim]
+}
+
+func (data *Float16VectorFieldData) GetRow(i int) interface{} {
+	return data.Data[i*data.Dim*2 : (i+1)*data.Dim*2]
+}
+
+var ErrWrongRowType = fmt.Errorf("Wrong row type")
```
Comment (jiaoew1991) [inline internal/storage/insert_data.go:289]: use merr

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**